alf/config_util.py

```diff
@@ -557,13 +561,21 @@ def _wrapper(*args, **kwargs):
                     unspecified_kw_args[name] = config.get_value()
                 config.set_used()
 
+        for key in guard_kwargs:
+            if key in kwargs and key in configs:
+                if kwargs[key] != configs[key].get_default_value():
+                    raise ValueError(
+                        f"The kwarg '{key}' of {fn.__qualname__} is guarded but has been modified. "
+                        f"Most likely partial() was used to change this value, which is not allowed."
+                    )
+
         return fn(*args, *unspecified_positional_args, **kwargs,
                   **unspecified_kw_args)
 
     return _wrapper
 
 
-def _decorate(fn_or_cls, name, whitelist, blacklist):
+def _decorate(fn_or_cls, name, whitelist, blacklist, guard_kwargs):
```

Comment:
It's not actually 'kwargs'? It's better that we also have this guard for positional arg?

Reply:
I've expanded the feature for positional args as well.